### DISCUSSION TITLE
When will the ITMS-90809 workaround be enabled by default, without experimental flag?

### DISCUSSION THREAD

**Discussion by johannes-keinestam** [xamarin/Xamarin.Forms#11561] [General] [answered]
Hi!

Saw that the ITMS-90809 AppStore warning has been fixed by allowing you to change the behavior of the Xamarin.iOS linker by enabling an experimental flag to the build. That's great, but are there any updates regarding when this will be enabled by default? I'm assuming well in time for the December 2020 cut-off date?

We'd quite like to avoid adding temporary experimental flags to our app code, so we've held off on actually enabling it -- but of course, if we must do at some point anyway we might as well enable it now.

Thanks!

**Comment by AmrAlSayed0** [comment, marked as answer]
Add this ```--optimize=force-rejected-types-removal,experimental-xforms-product-type --warn-on-type-ref=UIKit.UIWebView``` to your ```MtouchExtraArgs``` in the ```csproj``` file. and make sure you're on Xamarin.Forms 4.6 or higher

**Reply by johannes-keinestam** [reply]
Thanks for the reply! I'm well-aware of this -- but am just weary of enabling this (obviously, still, experimental) flag.

I assume that the flag will be more or less immediately deprecated once the December 2020 date passes, since it should reasonably be enabled by default in the Xamarin.Forms release that is live at that point. I'd much prefer not cluttering up the project with (presumably) temporary stuff like this.

**Reply by AmrAlSayed0** [reply]
As far as I recall, the only thing that this flag does is strip out all references of ```UIWebView``` which should be fine. Even if you use Forms' ```WebView``` in your app, ```Xamarin.Forms``` uses the new ```WkWebView```
https://github.com/xamarin/Xamarin.Forms/blob/main/Stubs/Xamarin.Forms.Platform.cs#L78-L81
Also I did some digging and I think the ```experimental-xforms-product-type``` flag is enabled by default (though I'm not sure)
https://github.com/xamarin/xamarin-macios/blob/main/tools/common/Optimizations.cs#L156-L159
So it seems that you only need the ```force-rejected-types-removal``` flag but I decided to put both anyway.

Edit:
I found this commit with the tags "xamarin-mac-6.22.1.8" "xamarin-mac-6.22.1.2" "xamarin-ios-13.22.1.8" "xamarin-ios-13.22.1.2"
https://github.com/xamarin/xamarin-macios/commit/c8a8fb31569a896b3d279b1d666c4303a2d79085
So probably any Xamarin.Mac or Xamrin.iOS higher than those versions will have ```experimental-xforms-product-type``` enabled by default

It is this PR https://github.com/xamarin/xamarin-macios/pull/8425 which is backported to d16-7 https://github.com/xamarin/xamarin-macios/pull/8427

**Reply by johannes-keinestam** [reply]
Great info, thanks! From what I can glean, the `force-rejected-types-removal` flag might even be required only if you use a linker setting other than "Do not link", so perhaps I would not even have to enable that. I'm assuming then if you run Xamarin.iOS >= 13.22.1.9 and Xamarin.Forms >= 4.6 you're good to go without changing any build settings, which answers my question very well!

**Reply by davidortinau** [reply]
It has been [proposed here](https://github.com/xamarin/Xamarin.Forms/issues/11857#issuecomment-677678828) to drop UIWebView from the next release (5.0). Please comment or react on the comment to register your feedback.
